### PR TITLE
client: added support for westend

### PR DESCRIPTION
**Pull request by Bullrich**
Created a new main page with a list of all the available chains.

Updated the content so it automatically assigns information based on the chain.

Added the network data object of the westend network.

Added a new page `/westend` where the chain is automatically set based on that information.

Added a network selection dropdown to switch between rococo and westend

---

This are all the steps you must follow to test it out:
- `git clone -b bullrich/client/westend git@github.com:paritytech/polkadot-testnet-faucet.git`
- `cd polkadot-testnet-faucet/client`
- `yarn install`
- `yarn dev`

After that go to [`localhost:5173`](http://localhost:5173) and you'll be able to visualize the website

### CODE REVIEW

File: client/tests/westend.test.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
+import {
```
Comment (rzadp) [inline client/tests/westend.test.ts:1]: Lots of duplication with `rococo.test.ts`, does it make sense to merge them into a single, parametrized set of tests?

File: client/static/FAQ.md [hunk 2]
```diff
@@ -4,21 +4,21 @@ This is a polkadot-testnet faucet. You can use this to load testnet tokens into
 
 ## What is a faucet?
 
-A faucet is a developer tool to get a small amount of testnet tokens (ROC) in order to test and troubleshoot a decentralized application or protocol before going live on Polkadot, where one must use real DOT.
+A faucet is a developer tool to get a small amount of testnet tokens (<NETWORK-TOKEN>) in order to test and troubleshoot a decentralized application or protocol before going live on Polkadot, where one must use real DOT.
```
Comment (rzadp) [inline client/static/FAQ.md:7]: In the rendered version, it disappears.

Screenshot 2023-06-05 at 10 26 26

Comment (rzadp) [reply client/static/FAQ.md:7]: How about simply skipping the symbol and talk about `tokens` only?

Comment (Bullrich) [reply client/static/FAQ.md:7]: That's weird. In both of my cases they appear:
image

image

Could you try rebuilding the project and trying again? Maybe it was a cache problem?

File: client/tests/westend.test.ts (hunk 1 above)
Comment (Bullrich) [reply client/tests/westend.test.ts:1]: You are right. I did it quite in a hurry, so let me fix that.

**Comment by Bullrich** [comment]
I still need to fix the `/` page as it has references to Rococo

File: client/static/FAQ.md (hunk 2 above)
Comment (rzadp) [reply client/static/FAQ.md:7]: Ah nevermind me, I was simply looking at the rendered markdown in github UI - forgot for a moment that it is processed by the framework.

All is good. :)

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -200,6 +200,15 @@ deploy-rococo-ink-docs:
...
 
+deploy-westend-web:
+  stage: deploy
+  <<: *deploy-prod-refs
+  extends: .deploy-with-argocd
+  environment: westend-web
+  variables:
+    DOMAIN: parity-testnet
+    APP: substrate-faucet-westend-web
+
```
Comment (mutantcornholio) [inline .gitlab-ci.yml:211]: let's drop this section, as we won't be deploying westend web faucet separately from its matrix counterpart

File: client/src/routes/westend/+page.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import type { PageLoad } from "./$types";
+
+export const load: PageLoad<{ faq: string }> = async ({ fetch }) => {
+	const data = await fetch("/FAQ.md");
+	const textData = await data.text();
```
Comment (mutantcornholio) [inline client/src/routes/westend/+page.ts:5]: Any particular reason not to include this on build and skip the fetch?

**Review by mutantcornholio — CHANGES_REQUESTED** [review]
LGTM, but let's update it as we decided to merge the faucets

**Review by mutantcornholio — CHANGES_REQUESTED** [review]
LGTM, but let's update it as we decided to merge the faucets

File: client/src/routes/westend/+page.ts (hunk 4 above)
Comment (Bullrich) [reply client/src/routes/westend/+page.ts:5]: Yeah, mainly to stop myself from installing 5 libraries that handle markdown parsing and import (too much boilerplate for a one particular case and I don't want to bloat the final js bundle).

By having it as a simple asset it is accesible and adds less loading time than an image.

Comment (Bullrich) [reply client/src/routes/westend/+page.ts:5]: Wait, let me try something native

Comment (Bullrich) [reply client/src/routes/westend/+page.ts]: It could be done easily and without any other libs or modifications! Brilliant!

Comment (mutantcornholio) [reply client/src/routes/westend/+page.ts]: Nice one!

**Comment by mordamax** [comment]
> You can leave an answer in this question on StackExchange when this PR is merged, to help the community know how to get Westend tokens: https://substrate.stackexchange.com/questions/8789/westend-westmint-difficulty-in-minting-a-fungible-asset-in-westmint

@AlexD10S this PR is unrelated to your issue
There are 2 other issues which should help you 
https://github.com/paritytech/polkadot-testnet-faucet/issues/152
https://github.com/paritytech/polkadot-testnet-faucet/issues/151

**Review by sergejparity — APPROVED** [review]

**Pull request merged**